docs/content/en/docs/fleet-manager/rollout/blue-green.md

```diff
@@ -0,0 +1,424 @@
+    - The `gateways` and `host` represent the ingress points for external and internal service traffic, respectively.
+- The `rolloutStatus` section displays the actual processing status of rollout within the fleet.
+
+About a minute after submitting this configuration, you can check the rollout status by running the following command:
+
+```conole
+kubectl get canary -n webapp --kubeconfig=/root/.kube/kurator-member1.config
+
+NAME      STATUS        WEIGHT   LASTTRANSITIONTIME
+backend   Initialized   0        2024-01-13T07:48:10Z
+```
+
+If the status shows as `Initialized`, it means the initialization of rollout process has completed successfully.
+
+**Notes**: In the above configuration, we set the `kustomization.interval` to 0s. This disables Fluxcd's periodic synchronization of configurations between the local mirror and cluster. The reason is that Flagger needs to modify the replica counts in Deployments to complete its initialization process. If you are uncertain whether the replicas for all applications in your deployments are set to zero, it is recommended to also set `kustomization.interval` to 0s.
+
+### Trigger Rollout
+
+A Blue/Green Deployment can be triggered by either updating the container image referenced in the git repository configuration, or directly updating the image of the deployment resource locally in the Kubernetes cluster.
+
+During the analysis you can generate HTTP 500 errors and high latency to test Kurator's rollback.
+
+Exec into the testlaoder pod to generate HTTP 500 errors and Generate latency.
+
+```console
+kubectl -n webapp exec -it backend-testloader-5f7bcd85bb-6bgdd sh
+
+watch curl http://backend-canary.webapp:9898/status/500
+
+watch curl http://backend-canary.webapp:9898/delay/1
+```
+
+Review the results:
+
+```console
+kubectl get canary -n webapp -w --kubeconfig=/root/.kube/kurator-member1.config
+
+NAME      STATUS        WEIGHT   LASTTRANSITIONTIME
+backend   Initialized   0        2024-01-13T08:06:10Z
+backend   Progressing   0        2024-01-13T08:10:40Z
+backend   Progressing   0        2024-01-13T08:12:10Z
+backend   Progressing   0        2024-01-13T08:13:40Z
+backend   Progressing   0        2024-01-13T08:15:10Z
+backend   Failed        0        2024-01-13T08:16:40Z
+backend   Failed        0        2024-01-13T08:18:10Z
+```
+
+{{< image width="100%"
+link="./image/blue-green.svg"
+>}}
+
+- As shown in the diagram, after triggering a Blue/Green Deployment, the Kurator Rollout Plugin will first create pod(s) for the new version.
+- During the testing period, all traffic will be incrementally routed to the Green version. A variety of testing metrics will be collected from this live environment validation.
+- If the number of errors or failures encountered during testing exceeds the `checkFailedTimes`, all traffic will automatically be rerouted back to the original stable version.
+- It will then remove the canary pod, completing the rollback process.
```

Comment:
will it replace the primary with canary?

And also in the diagram, i am not sure if the step 4 should be `v2`

Reply:
No, the canary would not replace the primary if the test fails. The diagram shows the process of handling a failed blue-green deployment, not a successful switch. So in step 4, the remaining pods would be of version v1, not v2.

It is only in the case of a successful blue-green deployment that the canary would replace the primary.

Reply:
Why donot we provide a example that can do blue-green deployment successfully